sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/models/DataLakeAclChangeFailedException.java

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.file.datalake.models;
+
+/**
+ * An exception thrown when an operation is interrupted and can be continued later on.
+ */
+public class DataLakeAclChangeFailedException extends RuntimeException {
```

Comment:
Should we make it checked exception? That would be compiler enforced reminder to add proper error handling for this api (during design review this was a hot topic - how to write resilient code for this api). Unless there's a guidance (I couldn't find any) or limitation (i.e. reactor doesn't like checked exceptions)

Reply:
Like do you mean defining the method to say " throws DataLakeAclChangeFailedException"?

Reply:
It's a good idea but it also doesnt match with all the other APIs where we throw RuntimeException for network errors

Reply:
`throws` and make it extend `Exception`. I'm not sure which one is more important - not standing out (which this API does anyway) or highlight that some error handling should be written to not miss ACL updates as this is SDK responsibility partially with current design. @alzimmermsft what do you think?

Reply:
Generally we have erred on using `Runtime` exceptions to prevent downstream code from requiring `try/catch` blocks everywhere. https://azure.github.io/azure-sdk/java_implementation.html#error-handling

There have been cases in the past where we do use checked exceptions, but those are generally to maintain exception types with a Java language interface.

@srnagar @JonathanGiles for a bit more input.

Reply:
Gotcha. So there is a guidance. Let's stick to it then.

Reply:
Agree with @alzimmermsft. We should be using appropriate derivatives of RuntimeException for most client library APIs to not force users to write try/catch blocks. Also, it is preferable to use one of the Java runtime exception types or exceptions defined in Azure core instead of creating new custom exceptions unless it is necessary to define a new type.

Reply:
I believe we were asked to create a new Exception type from guidance in .NET so we are just following suit here as well. The reason we need a custom type is because we need to expose the continuation token to the user.